src/js/users/components/__tests__/UserDetail.test.tsx

```diff
@@ -0,0 +1,280 @@
+            props.match.params.userId = userDetail.id;
+
+            const scope = mockApiGetUser(props.match.params.userId, userDetail);
+
+            renderWithRouter(<UserDetail {...props} />, {}, history);
+
+            expect(await screen.findByText("Change Password")).toBeInTheDocument();
+
+            expect(screen.getByText(userDetail.handle)).toBeInTheDocument();
+            expect(screen.getByLabelText("admin")).toBeInTheDocument();
+            expect(screen.getByText("Back To List")).toBeInTheDocument();
+
+            expect(screen.getByText("Force user to reset password on next login")).toBeInTheDocument();
+            expect(screen.getByText("Change Password")).toBeInTheDocument();
+
+            expect(await screen.findByText("Groups")).toBeInTheDocument();
+            expect(screen.getByText("Group0")).toBeInTheDocument();
+            expect(screen.getByText("Group1")).toBeInTheDocument();
+            expect(screen.getByText("Group2")).toBeInTheDocument();
+            expect(screen.getByText("Group3")).toBeInTheDocument();
+
+            expect(screen.getByText("Primary Group")).toBeInTheDocument();
+            expect(screen.getByRole("option", { name: "None" })).toBeInTheDocument();
+            expect(screen.getByRole("option", { name: "Group1" })).toBeInTheDocument();
+            expect(screen.getByRole("option", { name: "Group4" })).toBeInTheDocument();
+
+            expect(screen.getByText("Permissions")).toBeInTheDocument();
+            expect(screen.getByText("Change group membership to modify permissions")).toBeInTheDocument();
+            expect(screen.getByText("cancel_job")).toBeInTheDocument();
+            expect(screen.getByText("create_sample")).toBeInTheDocument();
+
+            scope.isDone();
+        });
+
+        it("should render loading when the user details hasn't loaded", () => {
+            renderWithRouter(<UserDetail {...props} />, {}, history);
+
+            expect(screen.getByLabelText("loading")).toBeInTheDocument();
+            expect(screen.queryByText("Groups")).not.toBeInTheDocument();
+            expect(screen.queryByText("bob")).not.toBeInTheDocument();
+        });
+
+        it("should render correctly when [administrator=false] and canModifyUser=false", async () => {
+            mockApiListGroups(groups);
+            mockGetAccountAPI(account);
+            const scope = mockApiGetUser(props.match.params.userId, userDetail);
+
+            renderWithRouter(<UserDetail {...props} />, {}, history);
+
+            expect(await screen.findByText("Change Password")).toBeInTheDocument();
+
+            expect(screen.queryByLabelText("admin")).not.toBeInTheDocument();
+            expect(screen.queryByText("User Role")).not.toBeInTheDocument();
+            expect(screen.getByText("Group4")).toBeInTheDocument();
+            expect(screen.getByText("create_sample")).toBeInTheDocument();
+
+            scope.isDone();
```

Comment:
```suggestion
            scope.done();
```
I think `scope.isDone()` just returns a bool indicating if all mocked scopes were called as expected. `scope.done()` raises an error if not everything has been called. Only essential if you expecting a mutation to have fired